File: Easy/Math/3658. GCD of Odd and Even Sums.cpp
// 3658. GCD of Odd and Even Sums ->app1
class Solution {
public:
    int gcdOfOddEvenSums(int n) {
        return n;
    }
};

//app_>2
class Solution {
public:
    int gcdOfOddEvenSums(int n) {
        int odd = pow(n,2);
        int even = n*(n+1);
        return gcd(odd,even);
    }
};

File: Easy/Math/3658. GCD of Odd and Even Sums.py
# 3658. GCD of Odd and Even Sums
class Solution:
    import math
    def gcdOfOddEvenSums(self, n: int) -> int:
        odd_sum = n**2
        even_sum = n*(n+1)
        return math.gcd(odd_sum,even_sum)

        
